stdlib/printexc.ml

```diff
@@ -52,6 +52,19 @@ let use_printers x =
     | [] -> None in
   conv (Atomic.get printers)
 
+let walk_sum_type x =
```

Comment:
_(This is about the implementation.)_ I personally prefer `extensible_variant` over `sum_type` because the printer here does not work for all variant types (and in OCaml documentation the term "sum types" is not used). This comment also applies to the function `string_of_sum_type` below.